### PR TITLE
Experimental: allow specifying network interface on client/socket

### PR DESCRIPTION
**Pull request by ccahoon**

### CODE REVIEW

File: F53OSCSocket.m [hunk 1]
```diff
@@ -206,16 +208,19 @@ - (BOOL) isUdpSocket
 
 @synthesize port;
 
+@synthesize interface;
+
 - (BOOL) startListening
 {
+            NSLog(@"listen with interaface: %@", self.interface);
```
Comment (ChrisAshworth) [inline F53OSCSocket.m]: leftover developer logging?

File: F53OSCClient.m [hunk 2]
```diff
@@ -184,6 +188,7 @@ - (NSDictionary *) state
 
 - (void) setState:(NSDictionary *)state
 {
+    self.interface = state[@"interface"];
```
Comment (ChrisAshworth) [inline F53OSCClient.m]: Question to double-check: does this magically come back nil if it was originally set as [NSNull null]

Comment (ccahoon) [inline F53OSCClient.m]: It magically comes back as nil or "", both of which I now handle.

**Comment by ccahoon** [comment]
merged elsewhere

File: F53OSCClient.m (hunk 2 above)
Comment (ChrisAshworth) [inline F53OSCClient.m]: We discussed this in-person and it's been fixed by handling @"" in various places rather [NSNull null].

**Pull request closed without merging**